virttest/utils_misc.py

```diff
@@ -3331,34 +3331,35 @@ class struct_timespec(ctypes.Structure):
         return monotonic_time_os()
 
 
-def verify_host_dmesg(dmesg_log_file=None, trace_re=None):
+def verify_host_dmesg(dmesg_log_file=None, ignore_result=False, level_check=3):
     """
     Find host call trace in dmesg log.
 
     :param dmesg_log_file: The file used to save host dmesg. If None, will save
                            host dmesg to logging.debug.
-    :param trace_re: re string used to filter call trace.
-    """
-    panic_re = [r"BUG:.*---\[ end trace .* \]---"]
-    panic_re.append(r"----------\[ cut here.* BUG .*\[ end trace .* \]---")
-    panic_re.append(r"general protection fault:.* RSP.*>")
-    panic_re.append(r"Out of memory: Kill process .*Call Trace:")
-    panic_re = "|".join(panic_re)
-    dmesg = process.system_output("dmesg", timeout=30, ignore_status=True,
-                                  verbose=False)
-    if dmesg:
-        match = re.search(panic_re, dmesg, re.DOTALL | re.MULTILINE | re.I)
-        if match is not None:
-            err = "Found call trace in host dmesg."
-            d_log = "Host dmesg log:\n%s" % dmesg
-            if dmesg_log_file:
-                file(dmesg_log_file, "w+").write(d_log)
-                err += " Please check host dmesg log %s." % dmesg_log_file
-            else:
-                err += " Please check host dmesg log in debug log."
-                logging.debug(d_log)
-            process.system("dmesg -C", ignore_status=True)
-            raise exceptions.TestError(err)
+    :param ignore_result: True or False, whether to fail test case on issues
+    :param level_check: level of severity of issues to be checked
+                        1 - emerg
+                        2 - emerg,alert
+                        3 - emerg,alert,crit
+                        4 - emerg,alert,crit,err
+                        5 - emerg,alert,crit,err,warn
+    """
+    cmd = "dmesg -T -l %s|grep ." % ",".join(map(str, range(1, int(level_check) + 1)))
```

Comment:
@ldoktor yes, you are right I am mistaken it starts from 1, actually it is from 0

```Supported log levels (priorities):
   emerg - system is unusable
   alert - action must be taken immediately
    crit - critical conditions
     err - error conditions
    warn - warning conditions
  notice - normal but significant condition
    info - informational
   debug - debug-level messages


For more details see dmesg(1).
 $ dmesg -l 3
[106187.835211] ipheth 2-1:4.2: ipheth_rcvbulk_callback: urb status: -71
[122630.127061] iwlwifi 0000:03:00.0: No association and the time event is over already...
[122632.724653] iwlwifi 0000:03:00.0: No association and the time event is over already...
[122635.968347] iwlwifi 0000:03:00.0: No association and the time event is over already...
[122639.919136] iwlwifi 0000:03:00.0: No association and the time event is over already...
[122643.692895] iwlwifi 0000:03:00.0: No association and the time event is over already...
[122647.447332] iwlwifi 0000:03:00.0: No association and the time event is over already...
[122651.683984] iwlwifi 0000:03:00.0: No association and the time event is over already...
[122654.934731] iwlwifi 0000:03:00.0: No association and the time event is over already...
 $ dmesg -l err
[106187.835211] ipheth 2-1:4.2: ipheth_rcvbulk_callback: urb status: -71
[122630.127061] iwlwifi 0000:03:00.0: No association and the time event is over already...
[122632.724653] iwlwifi 0000:03:00.0: No association and the time event is over already...
[122635.968347] iwlwifi 0000:03:00.0: No association and the time event is over already...
[122639.919136] iwlwifi 0000:03:00.0: No association and the time event is over already...
[122643.692895] iwlwifi 0000:03:00.0: No association and the time event is over already...
[122647.447332] iwlwifi 0000:03:00.0: No association and the time event is over already...
[122651.683984] iwlwifi 0000:03:00.0: No association and the time event is over already...
[122654.934731] iwlwifi 0000:03:00.0: No association and the time event is over already...

so 
 emerg - 0
   alert - 1
   crit - 2
   err - 3
   warn - 4
   notice - 5
    info - 6
   debug - 7```